pkg/controller/mizar/mizar-service-controller.go

```diff
@@ -213,115 +214,143 @@ func (c *MizarServiceController) syncService(eventKeyWithType KeyWithEventType)
 
 func (c *MizarServiceController) processServiceCreation(service *v1.Service, eventKeyWithType KeyWithEventType) error {
 	key := eventKeyWithType.Key
-	netName := getArktosNetworkName(service.Name)
-	networkName, hasDNSServiceLabel := service.Labels[arktosapisv1.NetworkLabel]
+	tenant, namespace, name, err := cache.SplitMetaTenantNamespaceKey(key)
+	if err != nil {
+		return err
+	}
+
+	// Get tenant default network
+	tenantDefaultNetwork, err := c.networkLister.NetworksWithMultiTenancy(tenant).Get(defaultNetworkName)
+	if tenantDefaultNetwork.Spec.Type != mizarNetworkType {
+		return nil
+	}
+
+	if tenantDefaultNetwork.Status.Phase != arktosapisv1.NetworkReady {
+		return errors.New(fmt.Sprintf("The arktos network %s is not Ready.", tenantDefaultNetwork.Name))
+	}
+
+	// update service spec with mizar annotation
+	obj, err := c.serviceLister.ServicesWithMultiTenancy(namespace, tenant).Get(name)
+	if err != nil {
+		return err
+	}
+	_, vpcNameOk := obj.Annotations[mizarAnnotationsVpcKey]
+	_, subnetNameOk := obj.Annotations[mizarAnnotationsSubnetKey]
 
-	if hasDNSServiceLabel && len(netName) != 0 {
-		klog.Infof("Starting ProcessArktosNetworkCreation network: %v", netName)
-		net, err := c.netClient.ArktosV1().NetworksWithMultiTenancy(service.Tenant).Get(netName, metav1.GetOptions{})
+	if !vpcNameOk && !subnetNameOk {
```

Comment:
probably should re-update if either one of them does not exist.

Reply:
If one of them does not exist, this is customer error. For 130, we will not deal with it.